### PR TITLE
Always unwind through a Resume and other fixes

### PR DESCRIPTION
**Pull request by arielb1**
Should fix most of the small MIR borrowck issues.

r? @nikomatsakis

### CODE REVIEW

**Comment by arielb1** [comment]
Ready for review

File: src/librustc_mir/borrow_check.rs [hunk 1]
```diff
@@ -710,6 +702,50 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
...
+
+        // FIXME: replace this with a proper borrow_conflicts_with_lvalue when
+        // that is merged.
+        let prefix_set = if might_be_alive {
+            PrefixSet::Supporting
+        } else {
+            PrefixSet::Shallow
+        };
+
+        self.prefixes(lvalue, prefix_set).any(|prefix| prefix == root_lvalue)
```
Comment (nikomatsakis) [inline src/librustc_mir/borrow_check.rs]: Tests?

**Review by nikomatsakis — APPROVED** [review]
Found the tests after all.

**Comment by nikomatsakis** [comment]
r=me post rebase

**Comment by kennytm** [comment]
Two of the new tests failed with WASM.

```
[01:38:26] ---- [mir-opt] mir-opt/match_false_edges.rs stdout ----
[01:38:26] 	thread '[mir-opt] mir-opt/match_false_edges.rs' panicked at 'Did not find expected line, error: Mismatch in lines
[01:38:26] Currnt block:  bb9: { // binding1 and guard
[01:38:26] Expected Line: "        _6 = const guard() -> bb10;      // scope 2 at /checkout/src/test/mir-opt/match_false_edges.rs:25:20: 25:27"
[01:38:26] Actual Line: "     _6 = const guard() -> [return: bb10, unwind: bb1];"
...
[01:38:26] ---- [mir-opt] mir-opt/validate_2.rs stdout ----
[01:38:26] 	thread '[mir-opt] mir-opt/validate_2.rs' panicked at 'Did not find expected line, error: Mismatch in lines
[01:38:26] Currnt block: None
[01:38:26] Expected Line: "        drop(_1) -> bb2;                 // scope 0 at /checkout/src/test/mir-opt/validate_2.rs:16:2: 16:2"
[01:38:26] Actual Line: "        drop(_1) -> [return: bb2, unwind: bb3];"
```

**Pull request merged**